spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java

```diff
@@ -57,13 +56,19 @@
  * @author Gary Russell
  *
  */
+@RabbitAvailable(queues = MessageListenerRecoveryRepeatIntegrationTests.TEST_QUEUE, purgeAfterEach = false)
+@LogLevels(level = "ERROR", lifecycle = Lifecycle.PER_CLASS, classes = { RabbitTemplate.class,
+		ConditionalRejectingErrorHandler.class,
+		SimpleMessageListenerContainer.class, BlockingQueueConsumer.class,
+		MessageListenerRecoveryRepeatIntegrationTests.class })
+@TestInstance(Lifecycle.PER_CLASS)
```

Comment:
Shouldn't we consider to have this option enabled globally for all the test classes?
I somehow believe that Spring Test Framework makes test instances really per class anyway...

Reply:
>I somehow believe

I didn't believe so (and I just tested it) - the context is cached but you get a new instance per test method.

We can try setting it globally, but I'd rather wait until all conversion is done before trying it.

Reply:
Yeah... Got it. Thank you!
Yep! Let's deffer it to the later phase!

Looking forward for your final solution for that `lifecycle` concern!